objectivec/ort_value.mm

```diff
@@ -92,6 +92,34 @@ - (nullable instancetype)initWithTensorData:(NSMutableData*)tensorData
   ORT_OBJC_API_IMPL_CATCH_RETURNING_NULLABLE(error)
 }
 
+- (nullable instancetype)initWithTensorStringData:(NSArray<NSString*>*)tensorStringData
+                                            shape:(NSArray<NSNumber*>*)shape
+                                            error:(NSError**)error {
+  try {
+    Ort::AllocatorWithDefaultOptions allocator;
+    const auto shapeVector = [shape]() {
+      std::vector<int64_t> result{};
+      result.reserve(shape.count);
+      for (NSNumber* dim in shape) {
+        result.push_back(dim.longLongValue);
+      }
+      return result;
+    }();
+    Ort::Value ortValue = Ort::Value::CreateTensor(
+        allocator, shapeVector.data(), shapeVector.size(), ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING);
+
+    size_t index = 0;
+    for (NSString *strinData in tensorStringData) {
```

Comment:
I think we should also ensure that the length of `tensorStringData` matches the tensor size. the C API `FillStringTensor` checks this or we can compute the size (there's `SafeMultiply` available here).